modules/ml/include/opencv2/ml.hpp

```diff
@@ -1496,6 +1496,187 @@ class CV_EXPORTS_W LogisticRegression : public StatModel
+  You will have to find the best termination criteria for your problem.
+
+Note that the parameters margin regularization, initial step size, and step decreasing power should be positive.
+
+To use SVMSGD algorithm do as follows:
+
+- first, create the SVMSGD object.
+
+- then set parameters  (model type, margin type, margin regularization, initial step size, step decreasing power) using the functions
+  setSvmsgdType(), setMarginType(), setMarginRegularization(), setInitialStepSize(), and setStepDecreasingPower(), or the function setOptimalParameters().
+
+- then the SVM model can be trained using the train features and the correspondent labels by the method train().
+
+- after that, the label of a new feature vector can be predicted using the method predict().
+
+@code
+// Create empty object
+cv::Ptr<SVMSGD> svmsgd = SVMSGD::create();
+
+// Set parameters
+svmsgd->setOptimalParameters();
+
+// Train the Stochastic Gradient Descent SVM
+SvmSgd->train(trainData);
+
+// Predict labels for the new samples
+svmsgd->predict(samples, responses);
+@endcode
+
+*/
+
+class CV_EXPORTS_W SVMSGD : public cv::ml::StatModel
+{
+public:
+
+    /** SVMSGD type.
+    ASGD is often the preferable choice. */
+    enum SvmsgdType
+    {
+        SGD, //!< Stochastic Gradient Descent
+        ASGD //!< Average Stochastic Gradient Descent
+    };
+
+    /** Margin type.*/
+    enum MarginType
+    {
+        SOFT_MARGIN, //!< General case, suits to the case of non-linearly separable sets, allows outliers.
+        HARD_MARGIN  //!< More accurate for the case of linearly separable sets.
+    };
+
+    /**
+     * @return the weights of the trained model (decision function f(x) = weights * x + shift).
+    */
+    CV_WRAP virtual Mat getWeights() = 0;
+
+    /**
+     * @return the shift of the trained model (decision function f(x) = weights * x + shift).
+    */
+    CV_WRAP virtual float getShift() = 0;
+
+
+    /** @brief Creates empty model.
+        Use StatModel::train to train the model. Since %SVMSGD has several parameters, you may want to
+        find the best parameters for your problem or use setOptimalParameters() to set some default parameters.
+    */
+    CV_WRAP static Ptr<SVMSGD> create();
+
+    /** @brief Function sets optimal parameters values for chosen SVM SGD model.
+     * If chosen type is ASGD, function sets the following values for parameters of model:
+     * marginRegularization = 0.00001;
+     * initialStepSize = 0.05;
+     * stepDecreasingPower = 0.75;
+     * termCrit.maxCount = 100000;
+     * termCrit.epsilon = 0.00001;
+     *
+     * If SGD:
+     * marginRegularization = 0.0001;
+     * initialStepSize = 0.05;
+     * stepDecreasingPower = 1;
+     * termCrit.maxCount = 100000;
+     * termCrit.epsilon = 0.00001;
+     * @param svmsgdType is the type of SVMSGD classifier. Legal values are SvmsgdType::SGD and SvmsgdType::ASGD.
+     * Recommended value is SvmsgdType::ASGD (by default).
```

Comment:
`SvmsgdType::SGD` is not valid for enum values access.
`SVMSGD::ASGD` looks better in [docs](http://pullrequest.opencv.org/buildbot/export/pr/6096/docs/de/d54/classcv_1_1ml_1_1SVMSGD.html)
Or just use `SVMSGD::SvmsgdType` for "name" of valid values

P.S. Same note for MarginType